src/protocolsupportpocketstuff/ProtocolSupportPocketStuff.java

```diff
@@ -38,7 +41,27 @@ public void onEnable() {
 		
 		pm("Hello world! :D");
 	}
-	
+
+	@EventHandler
+	public void onPlayerPropertiesResolve(PlayerPropertiesResolveEvent e) {
+		Connection con = e.getConnection();
+		if (PocketCon.isPocketConnection(con)) {
+			if (con.hasMetadata("applySkinOnJoin")) {
+				System.out.println("Applying cached for " + e.getConnection() + "...");
+				SkinUtils.SkinDataWrapper skinDataWrapper = (SkinUtils.SkinDataWrapper) con.getMetadata("applySkinOnJoin");
+				e.addProperty(new PlayerPropertiesResolveEvent.ProfileProperty("textures", skinDataWrapper.getValue(), skinDataWrapper.getSignature()));
+				con.removeMetadata("applySkinOnJoin");
+			}
+		}
+	}
+
+	@EventHandler
+	public void onConnectionOpen(ConnectionOpenEvent e) {
+		Connection con = e.getConnection();
+		// We can't check if it is a PE player yet because it is too early to figure out
```

Comment:
Why modal packet work then? :S Maybe we should move everything to here or other way around.

Reply:
@7kasper because ConnectionOpenEvent is before the version is detected by ProtocolSupport, ConnectionHandshakeEvent is after the Handshake (Pocket Login)

Reply:
Hmm. This logic will be changed in the future.. Will look into it.